pkg/registries/registries.go

```diff
@@ -24,6 +60,23 @@ type mirrorSet struct {
 	mirrorSourcePolicy config.MirrorSourcePolicy
 }
 
+// newMirror constructs an object with proper source and mirror name structures to be used in containerd registry config
+func newMirrorSet(srcImage string, mirrorLocations []config.ImageMirror, resolveTags bool,
+	mirrorSourcePolicy config.MirrorSourcePolicy) mirrorSet {
+	truncatedMirrors := []mirror{}
+	for _, m := range mirrorLocations {
+		truncatedMirrors = append(truncatedMirrors, newMirror(srcImage, string(m), resolveTags))
+	}
+	return mirrorSet{source: extractHostname(srcImage), mirrors: truncatedMirrors, mirrorSourcePolicy: mirrorSourcePolicy}
+}
+
+// extractHostname extracts just the initial host repo from a full image location
+// e.g. mcr.microsoft.com would be extracted from mcr.microsoft.com/oss/kubernetes/pause:3.9
+func extractHostname(fullImage string) string {
+	parts := strings.Split(fullImage, "/")
```

Comment:
should this account for the protocol in the full image URL? for example: `https://mcr.microsoft.com/oss/kubernetes/pause:3.9`

Reply:
Nope, having the web protocol in front of the repo is not accepted by the IDMS/ITMS resource so we'd never have to process such a URL